ip/ffmpeg.c

```diff
@@ -248,15 +250,15 @@ static int ffmpeg_open(struct input_plugin_data *ip_data)
 	av_opt_set_int(swr, "out_channel_layout", av_get_default_channel_layout(cc->channels), 0);
 #endif
 	av_opt_set_int(swr, "in_sample_rate",     cc->sample_rate, 0);
-	av_opt_set_int(swr, "out_sample_rate",    cc->sample_rate, 0);
+	av_opt_set_int(swr, "out_sample_rate",    out_sample_rate, 0);
 	av_opt_set_sample_fmt(swr, "in_sample_fmt",  cc->sample_fmt, 0);
 	priv->swr = swr;
 
 	ip_data->private = priv;
 #if LIBAVCODEC_VERSION_MAJOR >= 58
-	ip_data->sf = sf_rate(cc->sample_rate) | sf_channels(cc->ch_layout.nb_channels);
+	ip_data->sf = sf_rate(out_sample_rate) | sf_channels(cc->ch_layout.nb_channels);
 #else
-	ip_data->sf = sf_rate(cc->sample_rate) | sf_channels(cc->channels);
+	ip_data->sf = sf_rate(out_sample_rate) | sf_channels(cc->channels);
 #endif
 	switch (cc->sample_fmt) {
 	case AV_SAMPLE_FMT_U8:
```

Comment:
By the way it looks like `default`/`AV_SAMPLE_FMT_S16` is the only one that actually plays back normally. Removing the other cases works but I'd like to understand it more.

(This is nothing to do with DSD or this change though. DSD is `AV_SAMPLE_FMT_FLTP` so it takes the `default` path without issue.)

Reply:
The issue with the `AV_SAMPLE_FMT_U8` and `AV_SAMPLE_FMT_S32` cases is the hardcoded sizeof(int16_t) in the buffer_used_len calculation.